database/database_functions.py

```diff
@@ -1,64 +1,96 @@
-    r_dct = {}
-    for room in rooms:
-        r_dct.update({room.r_name:room.r_type})
+    def __init__(self):
+        pass
 
-    return r_dct
+    def initialize(self, filename):
+        '''
+        The function initaializes the database through a new databse or an existing database.
+        '''
+        self.ENGINE = create_engine('sqlite:///'+filename)
+        BASE.metadata.create_all(self.ENGINE)
+        self.DBSESSION = sessionmaker(bind=self.ENGINE)
+        self.SESSIONS = self.DBSESSION()
 
-def get_allocations():
-    allocations = SESSIONS.query(Allocations).all()
-    al_dct = {}
-    for allocation in allocations:
-        al_dct.update({allocation.p_name:[allocation.o_name, allocation.ls_name]})
+    def get_persons(self, filename):
+        '''
+        The function get the person from database and add the person to a dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        persons = self.SESSIONS.query(Persons).all()
+        person_dictionary = {}
+        for person in persons:
+            person_dictionary.update({person.person_name:person.person_type})
+        return person_dictionary
 
-    return al_dct
+    def get_rooms(self, filename):
+        '''
+        The function get the rooms from database and add the rooms to the dictionary with regards
+        to room name and room type to a dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        rooms = self.SESSIONS.query(Rooms).all()
+        room_dictionary = {}
+        for room in rooms:
+            room_dictionary.update({room.room_name:room.room_type})
+        return room_dictionary
 
+    def get_allocations(self, filename):
+        '''
+        The function get the allocations from database and adds the allocations to the dictionary
+        with regards to person name and person type and the room they belong to into a dictionary.
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        allocations = self.SESSIONS.query(Persons).all()
+        allocations_dictionary = {}
+        for allocation in allocations:
+            allocations_dictionary.update({allocation.person_name:[allocation.office_name,
+                                                                   allocation.livingspace_name]})
+        return allocations_dictionary
 
-def add_person(person_name, person_type):
-    ''' Adding a person to database'''
-    newperson = Persons(p_name=person_name, p_type=person_type)
-    SESSIONS.add(newperson)
-    SESSIONS.commit()
+    def add_person(self, filename, person_name, person_type, office_name, livingspace_name):
+        '''
+        The function is used to add the person to the dictionary
+        '''
+        # calling the initialize function to use the specified the database
```

Comment:
This comment is not really necessary, because there is a docstring explaining what the function does